charts/ChartConfig.tsx

```diff
@@ -696,8 +696,10 @@ export class ChartConfig {
             this.props.tab === "table" &&
             value === "chart" &&
             !this.userHasSetTimeline
-        )
+        ) {
             this.revertAutoSelectedStartYear()
```

Comment:
Would it be better to isolate this `this.revertAutoSelectedStartYear()` method in the data table class? Not sure how exactly, but maybe in something like `component...`methods?

Reply:
Yes, I'd also like to get some isolation here, but I'm not sure I understand what you mean by `component...` methods.